cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -304,8 +304,16 @@ void Node::AddOdometry(int64 timestamp, const string& frame_id,
         Eigen::Matrix3d::Zero(), rotational;
     // clang-format on
   }
-  map_builder_.GetTrajectoryBuilder(kTrajectoryBuilderId)
-      ->AddOdometerPose(time, pose, applied_covariance);
+  try {
+    const Rigid3d sensor_to_tracking =
+        LookupToTrackingTransformOrThrow(time, odometry.child_frame_id);
```

Comment:
Since you only use the "child_frame_id": Pass this as "frame_id" and remove the extra frame_id in the SensorData::Odometry struct?